commons/src/main/java/org/infinispan/commons/dataconversion/BinaryEncoder.java

```diff
@@ -53,7 +53,7 @@ private static boolean isTypeExcluded(Class<?> type) {
    }
 
    private boolean skipEncoding(Object source) {
-      return isTypeExcluded(source.getClass());
+      return source == null || isTypeExcluded(source.getClass());
```

Comment:
I think null should be handled separately and should never be presented to the Encoder.